cypress/integration/deckBuilder/dryRun.js

```diff
@@ -188,4 +209,190 @@ describe('Deck Builder — Dry-run', () => {
+
+      .drDrawHand(HAND)
+
+      .drEndTurn()
+      .drEndTurn()
+      .drEndTurn()
+      .drEndTurn()
+      .drEndTurn()
+      .drEndTurn()
+      .drEndTurn()
+
+      .drPlay('N22')
+
+      .get(s.DR_CARD)
+      .should('have.length', 3)
+
+      .drPlay('N12')
+
+      .get(s.DR_CARD)
+      .should('have.length', 1)
+
+      .drPlay('N14')
+
+      .get(s.DR_CARD)
+      .should('have.length', 2)
+
+      .drCycle(0)
+  })
+
+  it('should be possible to discard the whole hand with Snake Eyes', () => {
+    const HAND = ['N1', 'N2', 'N3', 'N33']
+
+    cy.visit(`/deck/${SNAKE_EYES_DECK}/dry-run?mode=MANUAL`)
+
+      .drDrawHand(HAND)
+
+      .drEndTurn()
+      .drEndTurn()
+
+      .drPlay('N33')
+
+      .get(s.DR_CARD)
+      .should('have.length', 4)
+  })
+
+  it('should not be possible to discard whole hand with Snake Eyes if hand is not full', () => {
+    const HAND = ['N1', 'N2', 'N3', 'N33']
+
+    cy.visit(`/deck/${SNAKE_EYES_DECK}/dry-run?mode=MANUAL`)
+
+      .drDrawHand(HAND)
+
+      .drEndTurn()
+      .drEndTurn()
+      .drEndTurn()
+      .drEndTurn()
+
+      .drPlay('N1')
+
+      .drPlay('N33')
+
+      .get(s.DR_CARD)
+      .should('have.length', 2)
+
+      .drPlay('N2')
+      .drCycle('N3')
+  })
+
+  it('should be possible to gain/spend mana', () => {
+    const HAND = ['N14', 'W10', 'W12', 'W19']
+
+    cy.visit(`/deck/${MANA_DECK}/dry-run?mode=MANUAL`)
+
+      .drDrawHand(HAND)
+
+      .drEndTurn()
+      .drEndTurn()
+      .drEndTurn()
+      .drEndTurn()
+
+      .drPlay('W19')
+
+      .get(s.DR_MANA)
+      .should('contain', 12)
+
+      .drPlay('W12')
+
+      .get(s.DR_MANA)
+      .should('contain', 10)
+
+      .drPlay('W10')
+
+      .get(s.DR_MANA)
+      .should('contain', 0)
+  })
+
+  it('should not be possible to play anything else than spells with Archdruid Earyn', () => {
+    const HAND = ['N1', 'N3', 'N4', 'N48']
+
+    cy.visit(`/deck/${EARYN_DECK}/dry-run?mode=MANUAL`)
+
+      .drDrawHand(HAND)
+
+      .drEndTurn()
+      .drEndTurn()
+      .drEndTurn()
+      .drEndTurn()
+
+      .drPlay('N48')
+
+      .get(s.DR_CARD)
+      .should('have.length', 3)
+  })
+
+  it('should be possible to play spells with Archdruid Earyn', () => {
+    const HAND = ['W1', 'N44', 'W19', 'N48']
+
+    cy.visit(`/deck/${EARYN_DECK}/dry-run?mode=MANUAL`)
+
+      .drDrawHand(HAND)
+
+      .drEndTurn()
+      .drEndTurn()
+      .drEndTurn()
+      .drEndTurn()
+
+      .drPlay('N48')
+
+      .get(s.DR_MANA)
+      .should('contain', 13)
+
+      .get(s.DR_CARD)
+      .should('have.length', 1)
+  })
+
+  it('should not be possible to play Icicle Burst with Archdruid Earyn with no frozen enemies', () => {
+    const HAND = ['W1', 'W2', 'W19', 'N48']
+
+    cy.visit(`/deck/${EARYN_DECK}/dry-run?mode=MANUAL`)
+
+      .drDrawHand(HAND)
+
+      .drEndTurn()
+      .drEndTurn()
+      .drEndTurn()
+      .drEndTurn()
+
+      .drPlay('N48')
+
+      .drPlay('W2')
+
+      .drPlay('W1')
+  })
```

Comment:
I'm getting an error in cypress after this last test, in the afterEach function. Maybe because the four cards are played it can't find one to deselect?